ccdproc/tests/test_ccdproc.py

```diff
@@ -771,3 +771,71 @@ def test_wcs_project_onto_scale_wcs(ccd_data):
     # pixels outside the footprint of the original image, are the only
     # pixels that should be masked.
     assert new_ccd.mask.sum() == 4 + np.isnan(new_ccd.data).sum()
+
+
+def test_ccd_process_does_not_change_input(ccd_data):
+    original = ccd_data.copy()
+    ccd = ccd_process(ccd_data, gain=5 * u.electron / u.adu,
+                      readnoise=10 * u.electron)
+    np.testing.assert_array_equal(original.data, ccd_data.data)
+    assert original.unit == ccd_data.unit
+
+
+def test_ccd_process_parameters_are_aprropriate(ccd_data):
+    # oscan check
+    with pytest.raises(TypeError):
+        ccd_process(ccd_data, oscan=True)
+
+    # trim secion check
+    with pytest.raises(TypeError):
+        ccd_process(ccd_data, trim=True)
+
+    # error frame check
+    # gain and readnoise must be specified
+    with pytest.raises(ValueError):
+        ccd_process(ccd_data, error=True)
+
+    # gain must be specified
+    with pytest.raises(ValueError):
+        ccd_process(ccd_data, error=True, gain=None, readnoise=5)
+
+    # mask check
+    with pytest.raises(TypeError):
+        ccd_process(ccd_data, bad_pixel_mask=3)
+
+    # master bias check
+    with pytest.raises(TypeError):
+        ccd_process(ccd_data, master_bias=3)
+
+    # master flat check
+    with pytest.raises(TypeError):
+        ccd_process(ccd_data, master_flat=3)
+
+
+def test_ccd_process():
```

Comment:
Could you add a test in which either the master bias or flat is not a CCDData to make sure that works? Not critical...I just don't remember whether arithmetic works with a non-CCDData array.